### PR TITLE
User Auth Service

### PR DESCRIPTION
**Pull request by stephenprater**
## What changes does this PR introduce?

* Moves User Services into separate files.  Public classes should have their own files.
* Enables Local Operation of Holepunch, including deploys and a few QOL improvements like Projection Navigation
* Let's SSHD run under Trickle, which limits the bandwidth available to the process.

## Any background context you want to provide?

So the _actual_ feature is running under Trickle.  Sweeping enough floors to enable testing it locally is the bulk of the actual diff.

## Where should the reviewer start?

If you care about the housekeeping stuff, those are in the first two commits - if you only want to see the trickle stuff, that's in the last commit.

## Has this been manually tested? How?

Locally - all tests pass, I can run the project locally, and Trickle limits my bandwidth the something close the advertised number.

## What value does this provide to our end users?

Beta and Paid users will not have their usage adversely impacted by free-tier users hogging bandwidth.

## What GIF best describes this PR or how it makes you feel?
a pipe

### CODE REVIEW

**Comment by stephenprater** [comment]
[Holepunch Plan Authorization Service](https://trello.com/c/2mHJ7b20/343-holepunch-plan-authorization-service)

**Comment by stephenprater** [comment]
'bout time.

File: .env [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 DATABASE_URL=postgres@0.0.0.0:5432
 CONSUL_HOST=consul
 REDIS_HOST=redis
-SEA_HOST=nomad
-BASE_SERVICE_URL=localhost:5000
+SEA_HOST=172.16.123.1
+BASE_SERVICE_URL=holepunch.box
 SERVER_NAME=localhost:5000
 MIN_CALVER=2019.04.18.2
+SSH_KEY=~/.ssh/id_rsa.pub
```
Comment (AndrewScibek) [inline .env]: Is this just for local testing?

File: app/__init__.py [hunk 2]
```diff
@@ -98,7 +105,7 @@ def init_rollbar():
             got_request_exception.connect(rollbar.contrib.flask.report_exception, app)
 
     @app.errorhandler(429)
-    def too_many_requests(e):
+    def too_many_requests(_):
         return json_api(TooManyRequestsError, ErrorSchema), 429
 
     @app.errorhandler(500)
```
Comment (AndrewScibek) [inline app/__init__.py]: I remove this in my pr but shouldnt have this for public release. Breaks json schema

File: app/utils/dns.py [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 import random
+import os
 
-from typing import Iterator, NamedTuple
+from typing import Iterator, NamedTuple, List
 
 
 resolver = dns.resolver.Resolver(configure=False)
-resolver.nameservers = [os.getenv("DNS_ADDR", "172.31.1.88")]
+resolver.nameservers = [os.environ.get("DNS_ADDR")]
```
Comment (AndrewScibek) [inline app/utils/dns.py]: Why were all these `os.getenv` changed to `os.environ.get`?

File: app/__init__.py (hunk 2 above)
Comment (AndrewScibek) [reply app/__init__.py]: This is about the 500 error btw

File: .env (hunk 1 above)
Comment (stephenprater) [reply .env]: yes, docker only.

File: app/utils/dns.py (hunk 3 above)
Comment (stephenprater) [reply app/utils/dns.py]: https://stackoverflow.com/questions/16924471/difference-between-os-getenv-and-os-environ-get

TL;DR - If this stuff is not set, raise an exception.
